hudi-common/src/main/java/org/apache/hudi/io/storage/HoodieParquetStreamWriter.java

```diff
@@ -52,11 +53,23 @@ public HoodieParquetStreamWriter(FSDataOutputStream outputStream,
         .build();
   }
 
-  public void writeAvro(String key, R object) throws IOException {
-    writer.write(object);
+  @Override
+  public boolean canWrite() {
+    return true;
+  }
+
+  @Override
+  public void writeAvro(String key, IndexedRecord record) throws IOException {
+    writer.write(record);
     writeSupport.add(key);
   }
 
+  @Override
+  public void writeAvroWithMetadata(HoodieKey key, IndexedRecord avroRecord) throws IOException {
+    // TODO support populating the metadata
```

Comment:
doing it in this PR?

Reply:
HoodieParquetStreamWriter#writeAvroWithMetadata is not used in the code. We can implement later if we need it.